lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -704,4 +705,52 @@ export default class BitcoinProcessor {
     // non sidetree transaction
     return sidetreeTxToAdd;
   }
+
+  /**
+   * Return transactions since transaction number and number of blocks acquired.
+   * @param since Transaction number to query since
+   * @param numOfBlocks number of blocks worth of transactions to get at least
+   * @returns a tuple of [transactions, numberOfBlocksContainedInTransactions]
+   */
+  private async getAtLeastPageSizeBlocksWorthOfTransactionsSinceTransactionNumber (since: number | undefined): Promise<[TransactionModel[], number]> {
+    let beginTransactionTime = since === undefined ? this.genesisBlockNumber : TransactionNumber.getBlockNumber(since);
+    let endTransactionTime = beginTransactionTime + BitcoinProcessor.pageSizeInBlocks;
+    let numOfBlocksAcquired = 0;
+
+    const transactionsToReturn: TransactionModel[] = [];
+
+    // while need more blocks and have not reached the processed block
+    while (numOfBlocksAcquired < BitcoinProcessor.pageSizeInBlocks && beginTransactionTime <= this.lastProcessedBlock!.height) {
+      let transactions: TransactionModel[] = await this.transactionStore.getTransactionsByTransactionTime(beginTransactionTime, endTransactionTime);
```

Comment:
I am a bit concerned that with an `await` in the while loop, `this.lastProcessedBlock` would change as the loops goes on.